arbitrator/wasm-libraries/user-host/src/host.rs

```diff
@@ -39,14 +39,37 @@ pub unsafe extern "C" fn user_host__storage_store_bytes32(key: usize, value: usi
     let program = Program::start(2 * PTR_INK + EVM_API_INK);
     program.require_gas(evm::SSTORE_SENTRY_GAS).unwrap();
 
-    let api = &mut program.evm_api;
     let key = wavm::read_bytes32(key);
     let value = wavm::read_bytes32(value);
 
+    let api = &mut program.evm_api;
     let gas_cost = api.set_bytes32(key, value).unwrap();
     program.buy_gas(gas_cost).unwrap();
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn user_host__storage_transient_load_bytes32(key: usize, dest: usize) {
+    let program = Program::start(2 * PTR_INK + EVM_API_INK);
+    program.buy_gas(TRANSIENT_BYTES32_GAS).unwrap();
+
+    let key = wavm::read_bytes32(key);
+
+    let value = program.evm_api.transient_get_bytes32(key);
+    wavm::write_bytes32(dest, value);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn user_host__storage_transient_store_bytes32(key: usize, value: usize) {
+    let program = Program::start(2 * PTR_INK + EVM_API_INK);
+    program.buy_gas(TRANSIENT_BYTES32_GAS).unwrap();
+
+    let key = wavm::read_bytes32(key);
+    let value = wavm::read_bytes32(value);
+
+    let api = &mut program.evm_api;
+    api.transient_set_bytes32(key, value).unwrap();
```

Comment:
Could we move wavm::read_bytes32(key) and value into the `api.transient_set_bytes32(key, value).unwrap();` to avoid declaring two variables?

Reply:
This code has been replaced by macro.  In general, better to keep things explicit so that the amount of gas used is as clear as possible